src/controller/data_model/BUILD.gn

```diff
@@ -23,4 +23,7 @@ chip_data_model("data_model") {
 
   use_default_client_callbacks = true
   allow_circular_includes_from = [ "${chip_root}/src/controller" ]
+
+  # TODO: generated file dependency should be a lib NOT an include
+  include_dirs = [ "${chip_root}/zzz_generated/app-common" ]
```

Comment:
Use the one from src/ ?

Reply:
Done, removed the include_dirs support.

This was remaining from where I found that at least data_model has cross dependencies (all examples gen files have this too) and never removed it thinking 'good marker'. However this is easily changed when we fix all dependencies anyway.